padrino-core/lib/padrino-core/application/routing.rb

```diff
@@ -539,6 +542,12 @@ def route(verb, path, *args, &block)
 
         invoke_hook(:padrino_route_added, route, verb, path, args, options, block)
 
+        block_parameter_length = route.block_parameter_length
+        if block_arity.nonzero? && block_parameter_length != block_arity
```

Comment:
It should allow negative arity values that are not equal to the number of captures. ref: (http://ruby-doc.org/core-1.9.3/Proc.html#method-i-arity)[negative arity]